tests/app/delivery/test_send_to_providers.py

```diff
@@ -490,29 +488,19 @@ def test_should_call_send_sms_response_task_if_research_mode(
     send_to_providers.send_sms_to_provider(sample_notification)
     assert not aws_sns_client.send_sms.called
 
-    app.delivery.send_to_providers.send_sms_response.assert_called_once_with("sns", sample_notification.to)
+    app.delivery.send_to_providers.send_sms_response.assert_called_once_with(
+        "sns", sample_notification.to, sample_notification.reference
+    )
 
     persisted_notification = notifications_dao.get_notification_by_id(sample_notification.id)
     assert persisted_notification.to == sample_notification.to
     assert persisted_notification.template_id == sample_notification.template_id
     assert persisted_notification.status == "sent"
     assert persisted_notification.sent_at <= datetime.utcnow()
     assert persisted_notification.sent_by == "sns"
-    assert persisted_notification.reference == reference
     assert not persisted_notification.personalisation
 
 
-def test_should_not_have_sent_status_if_fake_callback_function_fails(sample_notification, mocker):
```

Comment:
This test was for the corner case where a user would send with a TEST key and somehow our fake callback code would fail (so the sms would stay CREATED and get resent). We're explicitly marking it SENT first now to make the callbacks more likely to pass the first time so this test no longer works.